src/server/scripts/Pet/pet_generic.cpp

```diff
@@ -320,10 +320,36 @@ class npc_pet_gen_mojo : public CreatureScript
         }
 };
 
+enum SoulTrader
+{
+    SAY_SOUL_TRADER_INTRO           = 0,
+
+    SPELL_PROC_TRIGGER_ON_KILL_AURA = 50051
+};
+
+struct npc_pet_gen_soul_trader : public ScriptedAI
+{
+    npc_pet_gen_soul_trader(Creature* creature) : ScriptedAI(creature) { }
+
+    void Reset() override
+    {
+        if (Unit* owner = me->GetOwner())
+            owner->RemoveAurasDueToSpell(SPELL_PROC_TRIGGER_ON_KILL_AURA);
```

Comment:
this is wrong.
This pet enter alot of times in evade, will make pet remove proc aura and all events of this pet depends of this proc.